From 3583791edd712abbc2c4066a9ab18813cf4d8aa0 Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <tmariyap@qti.qualcomm.com>
Date: Wed, 7 May 2025 23:05:30 +0530
Subject: [PATCH 2/5] UPSTREAM:DP:ath12k: Enable RAW MODE set bit flag

This patch enables the RAW mode set bit flag for RAW mode data transfer.

Original Patch:
000-d-Revert-wifi-ath12k-remove-unused-frame_mode-module-p.patch

Signed-off-by: Thiraviyam Mariyappan <tmariyap@qti.qualcomm.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index e23d1d4..ef72642 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -1341,6 +1341,9 @@ int ath12k_core_qmi_firmware_ready(struct ath12k_base *ab)
 		goto err_firmware_stop;
 	}
 
+	if (ath12k_frame_mode == ATH12K_HW_TXRX_RAW)
+		set_bit(ATH12K_FLAG_RAW_MODE, &ab->dev_flags);
+
 	mutex_lock(&ag->mutex);
 	mutex_lock(&ab->core_lock);
 
-- 
2.34.1

